### PR TITLE
[android] Remove few useless callbacks from RoutingController

### PR DESCRIPTION
**Pull request by rtsisyk**
Move few preliminary checks from RoutingController to MwmActivity.

### CODE REVIEW

File: android/app/src/main/java/app/organicmaps/routing/RoutingController.java [hunk 1]
```diff
@@ -905,12 +862,6 @@ public void setRouterType(@Framework.RouterType int router)
       build();
   }
 
-  @Framework.RouterType
-  public int getLastRouterType()
```
Comment (AndrewShkrob) [inline android/app/src/main/java/app/organicmaps/routing/RoutingController.java:909]: How can AA check that the already built route on the mobile phone is of type VEHICLE?

Comment (AndrewShkrob) [reply android/app/src/main/java/app/organicmaps/routing/RoutingController.java:909]: I've reviewed the logic, and this function is not necessary. Calling `setRouterType` is sufficient.

**Review by AndrewShkrob — APPROVED** [review]

**Comment by AndrewShkrob** [comment]
```java
default void showSearch() {}
```

This function is also not needed.

File: android/app/src/main/java/app/organicmaps/routing/RoutingController.java (hunk 1 above)
Comment (rtsisyk) [reply android/app/src/main/java/app/organicmaps/routing/RoutingController.java:909]: Ah, we can keep this method if it is needed for AA. I just removed all unused methods highlighted by Android Studio,

File: android/app/src/main/res/layout/dialog_suggest_reroute_title.xml
Comment (biodranik) [inline android/app/src/main/res/layout/dialog_suggest_reroute_title.xml:1]: Was it helpful for users? Do we need to restore it back?

**Review by biodranik — APPROVED** [review]

File: android/app/src/main/res/layout/dialog_suggest_reroute_title.xml
Comment (rtsisyk) [reply android/app/src/main/res/layout/dialog_suggest_reroute_title.xml:1]: I kept the same texts but using standard AlertDialog without custom layouts.

File: android/app/src/main/java/app/organicmaps/routing/RoutingController.java (hunk 1 above)
Comment (rtsisyk) [reply android/app/src/main/java/app/organicmaps/routing/RoutingController.java]: Restored.

**Comment by rtsisyk** [comment]
> ```java
> default void showSearch() {}
> ```
> 
> This function is also not needed.

Removed. Also removed showDownloader().

**Pull request merged**